### PR TITLE
feat(motion_velocity_planner,planning_evaluator): add  stop, slow_down diags

### PR DESCRIPTION
**Pull request by xtk8532704**
## Description

This PR adds a publisher in `motion_velocity_planner` to publish diagnostics as what [this PR](https://github.com/autowarefoundation/autoware.universe/pull/7960) did for `cruise_planner`.

The published diagnostics contain information that shows which module (DynamicObstacleStopModule; OutOfLaneModule; ObstacleVelocityLimiterModule) made what decision (stop; slow_down). An example is as follows:

```bash
$ ros2 topic echo /diagnostics
---
header:
  stamp:
    sec: 1723713886
    nanosec: 559327470
  frame_id: map
status:
- level: "\0"
  name: autoware::motion_velocity_planner::OutOfLaneModule.stop
  message: ''
  hardware_id: ''
  values:
  - key: decision
    value: none
- level: "\0"
  name: autoware::motion_velocity_planner::OutOfLaneModule.slow_down
  message: ''
  hardware_id: ''
  values:
  - key: decision
    value: autoware::motion_velocity_planner::OutOfLaneModule.slow_down
- level: "\0"
  name: autoware::motion_velocity_planner::ObstacleVelocityLimiterModule.stop
  message: ''
  hardware_id: ''
  values:
  - key: decision
    value: none
- level: "\0"
  name: autoware::motion_velocity_planner::ObstacleVelocityLimiterModule.slow_down
  message: ''
  hardware_id: ''
  values:
  - key: decision
    value: none
- level: "\0"
  name: autoware::motion_velocity_planner::DynamicObstacleStopModule.stop
  message: ''
  hardware_id: ''
  values:
  - key: decision
    value: none
- level: "\0"
  name: autoware::motion_velocity_planner::DynamicObstacleStopModule.slow_down
  message: ''
  hardware_id: ''
  values:
  - key: decision
    value: none
---
```
the decision info in diags will be used by [planning evaluator](https://github.com/autowarefoundation/autoware.universe/pull/7849).

## Related links

**Parent Issue:**

- Link

## How was this PR tested?
[Screencast from 2024年08月15日 18時24分11秒.webm](https://github.com/user-attachments/assets/dc0faf68-d040-4199-bd8a-aca99a56802e)

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by xtk8532704** [comment]
It is old PR：https://github.com/autowarefoundation/autoware.universe/pull/8492

**Comment by xtk8532704** [comment]
verified that `ros2 topic echo /planning/planning_evaluator/metrics` can got decision message of motion_velocity_planner from `planning_evaloator` side.

image

File: planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/planner_manager.cpp [hunk 1]
```diff
@@ -71,13 +71,61 @@ void MotionVelocityPlannerManager::update_module_parameters(
...
+    else
+      key_value.value = "none";
+    status->values.push_back(key_value);
+  }
+  // Add other information to the status if necessary in the future.
+  return status;
+}
+
+std::shared_ptr<DiagnosticArray> MotionVelocityPlannerManager::get_diagnostics(
+  const rclcpp::Time & current_time, const bool decided_diagnostics_only) const
```
Comment (kosuke55) [inline planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/planner_manager.cpp:95]: `decided_diagnostics_only` is always true now, is this necessary?

File: planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/planner_manager.cpp [hunk 2]
```diff
@@ -71,13 +71,61 @@ void MotionVelocityPlannerManager::update_module_parameters(
...
+{
+  auto status = std::make_shared<DiagnosticStatus>();
+  status->level = status->OK;
+  status->name = reason;
+  diagnostic_msgs::msg::KeyValue key_value;
+  {
+    // Decision
+    key_value.key = "decision";
+    if (is_decided)
+      key_value.value = reason;
```
Comment (kosuke55) [inline planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/planner_manager.cpp:85]: name and reason is same.  just `stop` or `slow_donw` is better to me

**Comment by xtk8532704** [comment]
Evaluator Test:  https://evaluation.tier4.jp/evaluation/reports/4d99b8c8-29c8-5edd-937e-eae4735e7901?project_id=prd_jt

**Review by kosuke55 — APPROVED** [review]
The code and behavior look good.
If you can confirm that there is no degradation in the evaluator, you can put it in the description and it is ok to merge.

**Comment by xtk8532704** [comment]
Test it again because the CommonScenario_basic is changed:https://evaluation.tier4.jp/evaluation/reports/b69972f4-0394-5d9b-9b54-a67847da7a79?project_id=prd_jt

**Comment by kosuke55** [comment]
@maxime-clem 
could you please check as conde owner?

**Review by maxime-clem — APPROVED** [review]
Code looks good to me

**Pull request merged**